Object name suffixes in .blend result in a crash in the x86_32 editor for Windows

### Tested versions

- Reproducible in: 4.7.1-stable (Windows - x86_32), 4.7-stable (Windows - x86_32)
- Not reproducible in: 4.7.1-stable (Windows - x86_64), 4.7-stable (Windows - x86_64)

### System information

Godot v4.7.1.stable - Windows 10 (build 19045) - Multi-window, 1 monitor - OpenGL 3 (Compatibility) - NVIDIA GeForce RTX 2060 (NVIDIA; 32.0.15.6094) - AMD Ryzen 7 4800H with Radeon Graphics (16 threads) - 3.87 GiB memory - WASAPI (48000 Hz, Stereo/mono)

### Issue description

Importing a `.blend` file with `-col` or `-convcol` suffixes (and likely other) in object names results in a crash when it's added to a scene or an inherited scene is created. This has been observed in x86_32 builds for Windows; 64-bit counterparts are OK. Other suffixes and platforms have not been tested.

```
================================================================
CrashHandlerException: Program crashed with signal 11
Engine version: Godot Engine v4.7.1.stable.official (a13da4feb8d8aefc283c3763d33a2f170a18d541)
Dumping the backtrace. Please include this when reporting the bug on: https://github.com/godotengine/godot/issues
Load address: 50000

[1] 4899177 (main+4449177) - no debug info in PE/COFF executable
[2] 760216d1 (kernelbase.dll+1e16d1) - no debug info in PE/COFF executable
-- END OF C++ BACKTRACE --
================================================================
```

### Steps to reproduce

1. Import a `.blend` with at least 1 suffix in an object's name, for example: `Cube-col`
2. Drag the `.blend` into a scene or create an inherited scene - the editor should crash.

### Minimal reproduction project (MRP)

This 4.7.1-stable project already contains `default_cube_with_collision.blend` for your convenience, simply add it to a scene or create an inherited scene. To edit the `.blend` file you will need the latest `5.2` version of Blender.

[gd_project_win_32_blend_import_crash.zip](https://github.com/user-attachments/files/30235861/gd_project_win_32_blend_import_crash.zip)

I can reproduce this with the 4.7.1.stable official x86_32 binary (which is compiled with MinGW).

I bisected the regression to https://github.com/godotengine/godot/pull/115877. cc @mihe @jrouwe 

When compiling 5902d3f37 locally with MSVC 2022 (tested 17.14.13 and 17.14.37) and `arch=x86_32 optimize=none` or `optimize=debug`, I can't reproduce the issue.

I can't compile with MSVC 2022 + `arch=x86_32 optimize=speed` or `optimize=speed_trace`:
```
C:\Users\Hugo\Documents\Git\godotengine\godot\thirdparty\astcenc\astcenc_weight_align.cpp(241) : fatal error C1001: Internal compiler error.
(compiler file 'D:\a\_work\1\s\src\vctools\Compiler\Utc\src\p2\main.cpp', line 258)
 To work around this problem, try simplifying or changing the program near the locations listed above.
If possible please provide a repro here: https://developercommunity.visualstudio.com
Please choose the Technical Support command on the Visual C++
 Help menu, or open the Technical Support help file for more information
```

```
Microsoft (R) C/C++ Optimizing Compiler Version 19.44.35228 for x86
```

However, I can reproduce the issue with MinGW32 using `optimize=speed_trace debug_symbols=yes separate_debug_symbols=yes` (GCC 16.1.0 from MSYS2). `separate_debug_symbols=yes` is needed for the build to complete (otherwise, it seems to go on forever).

<details>
<summary>gdb backtrace</summary>

```
#0  0x01b0aacd in _mm_min_ps(float __vector(4), float __vector(4)) (__A=..., __B=...)
    at C:/Users/Hugo/scoop/apps/msys2/2026-06-11/mingw32/lib/gcc/i686-w64-mingw32/16.1.0/include/xmmintrin.h:254
#1  JPH::Vec3::sMin (inV1=..., inV2=...) at thirdparty/jolt_physics/Jolt/Math/Vec3.inl:186
#2  JPH::AABox::Encapsulate (this=warning: (Error: pc 0x503d5588 in address map, but not in symtab.)
0x503d5588, inRHS=...) at thirdparty/jolt_physics/Jolt/Geometry/AABox.h:71
#3  JPH::AABBTreeBuilder::BuildInternal (this=this@entry=0xbfea08, inTriangles=...)
    at thirdparty/jolt_physics/Jolt/AABBTree/AABBTreeBuilder.cpp:221
#4  0x01b0aa94 in JPH::AABBTreeBuilder::BuildInternal (this=this@entry=0xbfea08, inTriangles=...)
    at thirdparty/jolt_physics/Jolt/AABBTree/AABBTreeBuilder.cpp:216
#5  0x01b0ad89 in JPH::AABBTreeBuilder::Build (this=this@entry=0xbfea08, outStats=...)
    at thirdparty/jolt_physics/Jolt/AABBTree/AABBTreeBuilder.cpp:160
#6  0x01a683b5 in JPH::MeshShape::MeshShape (this=warning: (Internal error: pc 0x47bb0a50 in read in CU, but not in symtab.)
warning: (Error: pc 0x47bb0a50 in address map, but not in symtab.)
0x47bb0a50, inSettings=..., outResult=...)
    at thirdparty/jolt_physics/Jolt/Physics/Collision/Shape/MeshShape.cpp:224
#7  0x01a68c8d in JPH::MeshShapeSettings::Create (this=this@entry=0xbfeb90)
    at thirdparty/jolt_physics/Jolt/Physics/Collision/Shape/MeshShape.cpp:119
#8  0x019e0d3a in JoltConcavePolygonShape3D::_build (this=warning: (Internal error: pc 0x48ba03b0 in read in CU, but not in symtab.)
warning: (Error: pc 0x48ba03b0 in address map, but not in symtab.)
0x48ba03b0)
    at modules/jolt_physics/shapes/jolt_concave_polygon_shape_3d.cpp:75
#9  0x019e798b in JoltShape3D::try_build (this=warning: (Internal error: pc 0x48ba03b0 in read in CU, but not in symtab.)
warning: (Error: pc 0x48ba03b0 in address map, but not in symtab.)
0x48ba03b0) at modules/jolt_physics/shapes/jolt_shape_3d.cpp:99
#10 0x019ea0c8 in JoltShapeInstance3D::try_build (this=this@entry=warning: (Internal error: pc 0x4d96e4a8 in read in CU, but not in symtab.)
warning: (Error: pc 0x4d96e4a8 in address map, but not in symtab.)
0x4d96e4a8)
    at modules/jolt_physics/shapes/jolt_shape_instance_3d.h:58
#11 0x019d7e57 in JoltShapedObject3D::_try_build_shape (this=this@entry=warning: (Internal error: pc 0x52cd1ea8 in read in CU, but not in symtab.)
warning: (Error: pc 0x52cd1ea8 in address map, but not in symtab.)
0x52cd1ea8, p_optimize_compound=true)
    at modules/jolt_physics/objects/jolt_shaped_object_3d.cpp:51
#12 0x019d81f7 in JoltShapedObject3D::build_shapes (this=warning: (Internal error: pc 0x52cd1ea8 in read in CU, but not in symtab.)
warning: (Error: pc 0x52cd1ea8 in address map, but not in symtab.)
0x52cd1ea8,
    p_optimize_compound=p_optimize_compound@entry=true) at modules/jolt_physics/objects/jolt_shaped_object_3d.cpp:284
#13 0x019cf7d9 in JoltBody3D::_add_to_space (this=warning: (Internal error: pc 0x52cd1ea8 in read in CU, but not in symtab.)
warning: (Error: pc 0x52cd1ea8 in address map, but not in symtab.)
0x52cd1ea8) at modules/jolt_physics/objects/jolt_body_3d.cpp:93
#14 0x019d3e0a in JoltObject3D::set_space (this=this@entry=warning: (Internal error: pc 0x52cd1ea8 in read in CU, but not in symtab.)
warning: (Error: pc 0x52cd1ea8 in address map, but not in symtab.)
0x52cd1ea8, p_space=<optimized out>)
    at modules/jolt_physics/objects/jolt_object_3d.cpp:95
#15 0x019aa744 in JoltPhysicsServer3D::body_set_space (this=<optimized out>, p_body=..., p_space=...)
    at modules/jolt_physics/jolt_physics_server_3d.cpp:548
#16 0x05ea23e4 in PhysicsServer3DWrapMT::body_set_space (this=warning: (Internal error: pc 0x38001b00 in read in CU, but not in symtab.)
warning: (Error: pc 0x38001b00 in address map, but not in symtab.)
0x38001b00, p1=..., p2=...)
    at servers/physics_3d/physics_server_3d_wrap_mt.h:182
#17 0x03b5a5d6 in CollisionObject3D::_notification (this=warning: (Internal error: pc 0x4d95f190 in read in CU, but not in symtab.)
warning: (Error: pc 0x4d95f190 in address map, but not in symtab.)
0x4d95f190, p_what=41)
    at scene/3d/physics/collision_object_3d.cpp:85
#18 0x04f29882 in Object::_notification_forward (this=this@entry=warning: (Internal error: pc 0x4d95f190 in read in CU, but not in symtab.)
warning: (Error: pc 0x4d95f190 in address map, but not in symtab.)
0x4d95f190, p_notification=p_notification@entry=41)
    at core/object/object.cpp:908
#19 0x03a7e6ed in Object::notification (this=warning: (Internal error: pc 0x4d95f190 in read in CU, but not in symtab.)
warning: (Error: pc 0x4d95f190 in address map, but not in symtab.)
0x4d95f190, p_notification=41, p_reversed=false)
    at ./core/object/object.h:730
#20 Node3D::_notification (this=warning: (Internal error: pc 0x4d95f190 in read in CU, but not in symtab.)
warning: (Error: pc 0x4d95f190 in address map, but not in symtab.)
0x4d95f190, p_what=<optimized out>) at scene/3d/node_3d.cpp:182
#21 0x05c4b488 in Node3D::_notification_forwardv (this=warning: (Internal error: pc 0x4d95f190 in read in CU, but not in symtab.)
warning: (Error: pc 0x4d95f190 in address map, but not in symtab.)
0x4d95f190, p_notification=10) at ./scene/3d/node_3d.h:51
#22 CollisionObject3D::_notification_forwardv (this=warning: (Internal error: pc 0x4d95f190 in read in CU, but not in symtab.)
warning: (Error: pc 0x4d95f190 in address map, but not in symtab.)
0x4d95f190, p_notification=10)
    at ./scene/3d/physics/collision_object_3d.h:40
#23 PhysicsBody3D::_notification_forwardv (this=warning: (Internal error: pc 0x4d95f190 in read in CU, but not in symtab.)
warning: (Error: pc 0x4d95f190 in address map, but not in symtab.)
0x4d95f190, p_notification=10)
    at ./scene/3d/physics/physics_body_3d.h:38
#24 StaticBody3D::_notification_forwardv (this=warning: (Internal error: pc 0x4d95f190 in read in CU, but not in symtab.)
warning: (Error: pc 0x4d95f190 in address map, but not in symtab.)
0x4d95f190, p_notification=10) at ./scene/3d/physics/static_body_3d.h:43
#25 0x04f29882 in Object::_notification_forward (this=this@entry=warning: (Internal error: pc 0x4d95f190 in read in CU, but not in symtab.)
warning: (Error: pc 0x4d95f190 in address map, but not in symtab.)
0x4d95f190, p_notification=p_notification@entry=10)
    at core/object/object.cpp:908
#26 0x034c773f in Object::notification (this=warning: (Internal error: pc 0x4d95f190 in read in CU, but not in symtab.)
warning: (Error: pc 0x4d95f190 in address map, but not in symtab.)
0x4d95f190, p_notification=10, p_reversed=false)
    at ./core/object/object.h:730
#27 Node::_propagate_enter_tree (this=warning: (Internal error: pc 0x4d95f190 in read in CU, but not in symtab.)
warning: (Error: pc 0x4d95f190 in address map, but not in symtab.)
0x4d95f190) at scene/main/node.cpp:360
#28 0x034c791d in Node::_propagate_enter_tree (this=warning: (Internal error: pc 0x5623f698 in read in CU, but not in symtab.)
warning: (Error: pc 0x5623f698 in address map, but not in symtab.)
0x5623f698) at scene/main/node.cpp:379
#29 0x034c791d in Node::_propagate_enter_tree (this=this@entry=warning: (Internal error: pc 0x4ac6c6a8 in read in CU, but not in symtab.)
warning: (Error: pc 0x4ac6c6a8 in address map, but not in symtab.)
0x4ac6c6a8) at scene/main/node.cpp:379
#30 0x034c7b7e in Node::_set_tree (this=warning: (Internal error: pc 0x4ac6c6a8 in read in CU, but not in symtab.)
warning: (Error: pc 0x4ac6c6a8 in address map, but not in symtab.)
0x4ac6c6a8, p_tree=warning: (Internal error: pc 0x380b3eb0 in read in CU, but not in symtab.)
warning: (Error: pc 0x380b3eb0 in address map, but not in symtab.)
0x380b3eb0) at scene/main/node.cpp:3399
#31 0x034c7c93 in Node::_add_child_nocheck (this=this@entry=warning: (Internal error: pc 0x3e221828 in read in CU, but not in symtab.)
warning: (Error: pc 0x3e221828 in address map, but not in symtab.)
0x3e221828, p_child=<optimized out>, p_name=...,
    p_internal_mode=Node::INTERNAL_MODE_DISABLED) at scene/main/node.cpp:1702
#32 0x034cbe20 in Node::add_child (this=this@entry=warning: (Internal error: pc 0x3e221828 in read in CU, but not in symtab.)
warning: (Error: pc 0x3e221828 in address map, but not in symtab.)
0x3e221828, rp_child=..., rp_child@entry=...,
    p_force_readable_name=<optimized out>, p_force_readable_name@entry=false, p_internal=<optimized out>,
    p_internal@entry=Node::INTERNAL_MODE_DISABLED) at scene/main/node.cpp:1732
#33 0x02620936 in EditorInterface::make_scene_preview (this=<optimized out>, p_path=..., p_scene=p_scene@entry=warning: (Internal error: pc 0x4ac6c6a8 in read in CU, but not in symtab.)
warning: (Error: pc 0x4ac6c6a8 in address map, but not in symtab.)

    0x4ac6c6a8, p_preview_size=p_preview_size@entry=1024) at ./core/object/object.h:1123
#34 0x02bcf98a in ResourceImporterScene::import (this=warning: (Internal error: pc 0x38211268 in read in CU, but not in symtab.)
warning: (Error: pc 0x38211268 in address map, but not in symtab.)
0x38211268, p_source_id=<optimized out>, p_source_file=...,
    p_save_path=..., p_options=..., r_platform_variants=<optimized out>, r_gen_files=<optimized out>,
    r_metadata=<optimized out>) at ./editor/editor_interface.h:97
#35 0x029daa77 in EditorFileSystem::_reimport_file (this=this@entry=warning: (Internal error: pc 0x38214f88 in read in CU, but not in symtab.)
warning: (Error: pc 0x38214f88 in address map, but not in symtab.)
0x38214f88, p_file=..., p_custom_options=...,
    p_custom_importer=..., p_generator_parameters=p_generator_parameters@entry=0x0,
    p_update_file_system=p_update_file_system@entry=true) at editor/file_system/editor_file_system.cpp:2928
#36 0x029e8c7b in EditorFileSystem::reimport_files (this=this@entry=warning: (Internal error: pc 0x38214f88 in read in CU, but not in symtab.)
warning: (Error: pc 0x38214f88 in address map, but not in symtab.)
0x38214f88, p_files=...)
    at editor/file_system/editor_file_system.cpp:3385
#37 0x029ee4f4 in EditorFileSystem::_update_scan_actions (this=this@entry=warning: (Internal error: pc 0x38214f88 in read in CU, but not in symtab.)
warning: (Error: pc 0x38214f88 in address map, but not in symtab.)
0x38214f88)
    at editor/file_system/editor_file_system.cpp:1048
#38 0x029ef34a in EditorFileSystem::_notification (this=warning: (Internal error: pc 0x38214f88 in read in CU, but not in symtab.)
warning: (Error: pc 0x38214f88 in address map, but not in symtab.)
0x38214f88, p_what=17)
    at editor/file_system/editor_file_system.cpp:1802
#39 0x04f29882 in Object::_notification_forward (this=this@entry=warning: (Internal error: pc 0x38214f88 in read in CU, but not in symtab.)
warning: (Error: pc 0x38214f88 in address map, but not in symtab.)
0x38214f88, p_notification=p_notification@entry=17)
    at core/object/object.cpp:908
#40 0x03506e56 in Object::notification (this=warning: (Internal error: pc 0x38214f88 in read in CU, but not in symtab.)
warning: (Error: pc 0x38214f88 in address map, but not in symtab.)
0x38214f88, p_notification=17, p_reversed=false)
    at ./core/object/object.h:730
#41 SceneTree::_process_group (this=this@entry=warning: (Internal error: pc 0x380b3eb0 in read in CU, but not in symtab.)
warning: (Error: pc 0x380b3eb0 in address map, but not in symtab.)
0x380b3eb0, p_group=<optimized out>, p_group@entry=warning: (Internal error: pc 0x380b3ff0 in read in CU, but not in symtab.)
warning: (Error: pc 0x380b3ff0 in address map, but not in symtab.)
0x380b3ff0,
    p_physics=p_physics@entry=false) at scene/main/scene_tree.cpp:1230
#42 0x035077aa in SceneTree::_process (this=this@entry=warning: (Internal error: pc 0x380b3eb0 in read in CU, but not in symtab.)
warning: (Error: pc 0x380b3eb0 in address map, but not in symtab.)
0x380b3eb0, p_physics=p_physics@entry=false)
    at scene/main/scene_tree.cpp:1307
#43 0x0350837a in SceneTree::process (this=warning: (Internal error: pc 0x380b3eb0 in read in CU, but not in symtab.)
warning: (Error: pc 0x380b3eb0 in address map, but not in symtab.)
0x380b3eb0, p_time=0.0033333333333333335) at scene/main/scene_tree.cpp:719
#44 0x00e4cfab in Main::iteration () at main/main.cpp:4958
#45 0x00df3473 in OS_Windows::run (this=this@entry=0xbffcb0) at platform/windows/os_windows.cpp:2359
#46 0x00e46693 in widechar_main (argc=2, argv=argv@entry=warning: (Internal error: pc 0xc72f540 in read in CU, but not in symtab.)
warning: (Error: pc 0xc72f540 in address map, but not in symtab.)
0xc72f540) at platform/windows/godot_windows.cpp:105
#47 0x00e466db in _main () at platform/windows/godot_windows.cpp:132
#48 0x0767a05a in main (warning: (Internal error: pc 0x75905d48 in read in CU, but not in symtab.)
warning: (Error: pc 0x75905d48 in address map, but not in symtab.)
argc=2, argv=warning: (Internal error: pc 0xc692f28 in read in CU, but not in symtab.)
warning: (Error: pc 0xc692f28 in address map, but not in symtab.)
0xc692f28) at platform/windows/godot_windows.cpp:151
#49 0x00df11ac in __tmainCRTStartup () at D:/W/B/src/mingw-w64/mingw-w64-crt/crt/crtexe.c:265
warning: (Internal error: pc 0x75905d48 in read in CU, but not in symtab.)
warning: (Error: pc 0x75905d48 in address map, but not in symtab.)
warning: (Internal error: pc 0x75905d48 in read in CU, but not in symtab.)
warning: (Error: pc 0x75905d48 in address map, but not in symtab.)
warning: (Internal error: pc 0x75905d48 in read in CU, but not in symtab.)
warning: (Error: pc 0x75905d48 in address map, but not in symtab.)
warning: (Internal error: pc 0x75905d48 in read in CU, but not in symtab.)
warning: (Error: pc 0x75905d48 in address map, but not in symtab.)
#50 0x75905d49 in KERNEL32!BaseThreadInitThunk () from C:\WINDOWS\SysWOW64\kernel32.dll
warning: (Internal error: pc 0x75905d48 in read in CU, but not in symtab.)
warning: (Error: pc 0x75905d48 in address map, but not in symtab.)
warning: (Internal error: pc 0x77e7e00a in read in CU, but not in symtab.)
warning: (Error: pc 0x77e7e00a in address map, but not in symtab.)
warning: (Internal error: pc 0x77e7e00a in read in CU, but not in symtab.)
warning: (Error: pc 0x77e7e00a in address map, but not in symtab.)
warning: (Internal error: pc 0x77e7e00a in read in CU, but not in symtab.)
warning: (Error: pc 0x77e7e00a in address map, but not in symtab.)
warning: (Internal error: pc 0x77e7e00a in read in CU, but not in symtab.)
warning: (Error: pc 0x77e7e00a in address map, but not in symtab.)
#51 0x77e7e00b in ntdll!RtlInitializeExceptionChain () from C:\WINDOWS\SysWOW64\ntdll.dll
warning: (Internal error: pc 0x77e7e00a in read in CU, but not in symtab.)
warning: (Error: pc 0x77e7e00a in address map, but not in symtab.)
warning: (Internal error: pc 0x77e7df90 in read in CU, but not in symtab.)
warning: (Error: pc 0x77e7df90 in address map, but not in symtab.)
warning: (Internal error: pc 0x77e7df90 in read in CU, but not in symtab.)
warning: (Error: pc 0x77e7df90 in address map, but not in symtab.)
warning: (Internal error: pc 0x77e7df90 in read in CU, but not in symtab.)
warning: (Error: pc 0x77e7df90 in address map, but not in symtab.)
warning: (Internal error: pc 0x77e7df90 in read in CU, but not in symtab.)
warning: (Error: pc 0x77e7df90 in address map, but not in symtab.)
#52 0x77e7df91 in ntdll!RtlGetAppContainerNamedObjectPath () from C:\WINDOWS\SysWOW64\ntdll.dll
warning: (Internal error: pc 0x77e7df90 in read in CU, but not in symtab.)
warning: (Error: pc 0x77e7df90 in address map, but not in symtab.)
#53 0x00000000 in ?? ()
```
</details>

--- a/thirdparty/README.md
+++ b/thirdparty/README.md
@@ -519,6 +519,10 @@ Files extracted from upstream source:
 - All files in `Jolt/`, except `Jolt/Jolt.cmake`, any files dependent on `ENABLE_OBJECT_STREAM` as seen in `Jolt/Jolt.cmake` and the `Jolt/Physics/Hair/`, `Jolt/Compute/` and `Jolt/Shaders/` folders.
 - `LICENSE`
 
+Patches:
+
+- `0001-backport-upstream-commit-63f2f57.patch` (GH-121974)
+
 
 ## libbacktrace
 

--- a/thirdparty/jolt_physics/Jolt/Core/Core.h
+++ b/thirdparty/jolt_physics/Jolt/Core/Core.h
@@ -585,7 +585,14 @@ static_assert(sizeof(uint64) == 8, "Invalid size of uint64");
 // Default memory allocation alignment.
 // This define can be overridden in case the user provides an Allocate function that has a different alignment than the platform default.
 #ifndef JPH_DEFAULT_ALLOCATE_ALIGNMENT
-	#define JPH_DEFAULT_ALLOCATE_ALIGNMENT __STDCPP_DEFAULT_NEW_ALIGNMENT__
+	#if defined(JPH_COMPILER_MINGW) && JPH_CPU_ARCH_BITS == 32
+		// 32-bit MinGW g++ lies that __STDCPP_DEFAULT_NEW_ALIGNMENT__ is 16, but it is actually 8.
+		// See: https://github.com/godotengine/godot/issues/121608#issuecomment-5142806330
+		// Note: See similar fix for the definition of JPH_INTERNAL_DEFAULT_ALLOCATE.
+		#define JPH_DEFAULT_ALLOCATE_ALIGNMENT 8
+	#else
+		#define JPH_DEFAULT_ALLOCATE_ALIGNMENT __STDCPP_DEFAULT_NEW_ALIGNMENT__
+	#endif
 #endif
 
 // Cache line size (used for aligning to cache line)

--- a/thirdparty/jolt_physics/Jolt/Core/Memory.h
+++ b/thirdparty/jolt_physics/Jolt/Core/Memory.h
@@ -38,6 +38,7 @@ JPH_EXPORT void RegisterDefaultAllocator();
 // It uses the non-aligned version, which on 32 bit platforms usually returns an 8 byte aligned block.
 // We therefore default to 16 byte aligned allocations when the regular new operator is used.
 // See: https://github.com/godotengine/godot/issues/105455#issuecomment-2824311547
+// Note: See similar fix for the definition of JPH_DEFAULT_ALLOCATE_ALIGNMENT
 #if defined(JPH_COMPILER_MINGW) && JPH_CPU_ARCH_BITS == 32
 	#define JPH_INTERNAL_DEFAULT_ALLOCATE(size) JPH::AlignedAllocate(size, 16)
 	#define JPH_INTERNAL_DEFAULT_FREE(pointer) JPH::AlignedFree(pointer)

--- a/thirdparty/jolt_physics/patches/0001-backport-upstream-commit-63f2f57.patch
+++ b/thirdparty/jolt_physics/patches/0001-backport-upstream-commit-63f2f57.patch
@@ -1,0 +1,32 @@
+diff --git a/thirdparty/jolt_physics/Jolt/Core/Core.h b/thirdparty/jolt_physics/Jolt/Core/Core.h
+index 30377deeff..10362b3225 100644
+--- a/thirdparty/jolt_physics/Jolt/Core/Core.h
++++ b/thirdparty/jolt_physics/Jolt/Core/Core.h
+@@ -585,7 +585,14 @@ static_assert(sizeof(uint64) == 8, "Invalid size of uint64");
+ // Default memory allocation alignment.
+ // This define can be overridden in case the user provides an Allocate function that has a different alignment than the platform default.
+ #ifndef JPH_DEFAULT_ALLOCATE_ALIGNMENT
+-	#define JPH_DEFAULT_ALLOCATE_ALIGNMENT __STDCPP_DEFAULT_NEW_ALIGNMENT__
++	#if defined(JPH_COMPILER_MINGW) && JPH_CPU_ARCH_BITS == 32
++		// 32-bit MinGW g++ lies that __STDCPP_DEFAULT_NEW_ALIGNMENT__ is 16, but it is actually 8.
++		// See: https://github.com/godotengine/godot/issues/121608#issuecomment-5142806330
++		// Note: See similar fix for the definition of JPH_INTERNAL_DEFAULT_ALLOCATE.
++		#define JPH_DEFAULT_ALLOCATE_ALIGNMENT 8
++	#else
++		#define JPH_DEFAULT_ALLOCATE_ALIGNMENT __STDCPP_DEFAULT_NEW_ALIGNMENT__
++	#endif
+ #endif
+ 
+ // Cache line size (used for aligning to cache line)
+diff --git a/thirdparty/jolt_physics/Jolt/Core/Memory.h b/thirdparty/jolt_physics/Jolt/Core/Memory.h
+index 5df9781dd5..bf98d4fe79 100644
+--- a/thirdparty/jolt_physics/Jolt/Core/Memory.h
++++ b/thirdparty/jolt_physics/Jolt/Core/Memory.h
+@@ -38,6 +38,7 @@ JPH_EXPORT void RegisterDefaultAllocator();
+ // It uses the non-aligned version, which on 32 bit platforms usually returns an 8 byte aligned block.
+ // We therefore default to 16 byte aligned allocations when the regular new operator is used.
+ // See: https://github.com/godotengine/godot/issues/105455#issuecomment-2824311547
++// Note: See similar fix for the definition of JPH_DEFAULT_ALLOCATE_ALIGNMENT
+ #if defined(JPH_COMPILER_MINGW) && JPH_CPU_ARCH_BITS == 32
+ 	#define JPH_INTERNAL_DEFAULT_ALLOCATE(size) JPH::AlignedAllocate(size, 16)
+ 	#define JPH_INTERNAL_DEFAULT_FREE(pointer) JPH::AlignedFree(pointer)

